src/virtualenv/create/via_global_ref/builtin/pypy/pypy2.py

```diff
@@ -112,7 +112,7 @@ def modules(cls):
 
     @classmethod
     def _shared_libs(cls):
-        return ["libpypy-c.dll"]
+        return ["libpypy-c.dll", "libffi-7.dll"]
```

Comment:
Wouldn't this need to be version limited to newer pypy versions?

Reply:
no, the code to copy ignores missing files

Reply:
Oh, 🤔 wonder why 😆

Reply:
https://github.com/pypa/virtualenv/blob/17caadc974cb2a24f70374525596cd1160579594/src/virtualenv/create/via_global_ref/builtin/pypy/common.py#L45-L49

Reply:
Will it work without this DLL? Should we make it hard dependency rather than soft that it is now?

Reply:
Currently PyPy does not have the extra DLL, so making it required will break backward comatibility.

Reply:
Currently this DLL is soft depdency (so assuming that's not a bug), if in the future this is not the case we should make it hard dependency and version guard the hard dependency part.

Reply:
It would be nice to avoid this code altogether in favor of a more robust solution

Reply:
That wasn't my question. Can we say a usecase where pypy would work without this DLL with the current less robust solution? If yes, we can keep it a soft dependency. If not, it must be hard depedency.

Reply:
I am not sure what you mean, sorry. PyPy v7.3.5 will not have the libffi-7.dll and will work without it. Once the branch to add the dll gets merged, any future PyPy versions will require the dll. I am hopeful that branch will be merged before the PyPy v7.3.6 release, but it is dependent on a new release of virtualenv with this PR or one similar.

Reply:
On the other hand, I can envision a scenario where a packager like conda uses some [OS-level utilities](https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order) to avoid copying the dll. In this case even though the dlls are required, they are not copied by this code into the new virtualenv.